src/main/java/com/avispl/symphony/dal/avdevices/power/middleatlantic/nexsys/MiddleAtlanticUPSCommunicator.java

```diff
@@ -0,0 +1,962 @@
+	 * @return A human-readable representation of the time in the format "X hour(s) Y minute(s)" or "0 minute(s)" if the value is zero.
+	 * @throws IllegalArgumentException If the input value cannot be converted to a valid time format.
+	 */
+	private String convertTime(String value) {
+		try {
+			long inputValue = Long.parseLong(value);
+			long hours = inputValue / 60;
+			long minutes = inputValue % 60;
+
+			StringBuilder result = new StringBuilder();
+
+			if (hours > 0) {
+				result.append(hours).append(" hour(s) ");
+			}
+			if (minutes > 0) {
+				result.append(minutes).append(" minute(s) ");
+			}
+
+			return result.length() == 0 ? "0 minute(s)" : result.toString();
+		} catch (Exception e) {
+			logger.error("Error while formatting date: " + e.getMessage(), e);
+			return UPSConstant.NONE;
+		}
+	}
+
+	/**
+	 * Converts a date from one date format to another date format.
+	 *
+	 * @param input The input date string to be converted.
+	 * @param inputFormat The format of the input date string.
+	 * @param outputFormat The desired output date format.
+	 * @return The date string converted to the specified output format, or "None" if the input is "None".
+	 * @throws IllegalArgumentException If the input date cannot be converted from the input format to the output format.
+	 */
+	private String convertDateFormat(String input, String inputFormat, String outputFormat) {
+		if (UPSConstant.NONE.equals(input)) {
+			return input;
+		}
+		try {
+			SimpleDateFormat inputDateFormat = new SimpleDateFormat(inputFormat);
+			Date date = inputDateFormat.parse(input);
+
+			SimpleDateFormat outputDateFormat = new SimpleDateFormat(outputFormat);
+			return outputDateFormat.format(date);
+		} catch (Exception e) {
+			logger.error("Error while convert date format: " + e.getMessage(), e);
+			return UPSConstant.NONE;
+		}
+	}
+
+	/**
+	 * check value is null or empty
+	 *
+	 * @param value input value
+	 * @return value after checking
+	 */
+	private String getDefaultValueForNullData(String value) {
+		return StringUtils.isNotNullOrEmpty(value) ? value : UPSConstant.NONE;
+	}
+
+	/**
+	 * This method is used to validate input config management from user
+	 */
+	private void convertConfigManagement() {
+		isConfigManagement = StringUtils.isNotNullOrEmpty(this.configManagement) && this.configManagement.equalsIgnoreCase(UPSConstant.TRUE);
+	}
+
+	/**
+	 * Add advancedControllableProperties if advancedControllableProperties different empty
+	 *
+	 * @param advancedControllableProperties advancedControllableProperties is the list that store all controllable properties
+	 * @param stats store all statistics
+	 * @param property the property is item advancedControllableProperties
+	 * @return String response
+	 * @throws IllegalStateException when exception occur
+	 */
+	private void addAdvanceControlProperties(List<AdvancedControllableProperty> advancedControllableProperties, Map<String, String> stats, AdvancedControllableProperty property, String value) {
```

Comment:
just a little naming comment - I believe it was supposed to be AdvancedControlProperties, not Advance :)